src/vec.rs

```diff
@@ -143,9 +143,8 @@ impl<'data, T: Send> IndexedParallelIterator for Drain<'data, T> {
 
             // Create the producer as the exclusive "owner" of the slice.
             let producer = {
-                // Get a correct borrow lifetime, then extend it to the original length.
```

Comment:
This seems to have been written intentionally so that the lifetime of slice would be constrained by the function body whereas now `slice` could produce any lifetime required by `DrainProducer::new`.  Is it possible that inference produces an invalid lifetime here as the removed comment seemed to suggest?

Reply:
It is _more likely_ that the new code produce a wrong lifetime when subjected to future changes, so I've pinned down the lifetime to be sure.